### PR TITLE
Add setting for experimental features

### LINKED ISSUE #534: Expose Experimental Features Setting
The language server now has support for experimental features users can opt into. The first of which is validation of a folder on file save.

https://github.com/hashicorp/terraform-ls/pull/340

We need to expose a setting on the `terraform-ls` setting called `experimentalFeatures` and within that allow setting `validateOnSave = true`. Then we must pass the `experimentalFeatures` object to the language server within the `initializationOptions` object

### PR DESCRIPTION
**Pull request by aeschright**
- Boolean: validateOnSave

Closes #534

### CODE REVIEW

**Comment by aeschright** [comment]
Here's what it looks like:
Screen Shot 2021-01-05 at 3 56 14 PM
Screen Shot 2021-01-05 at 3 56 37 PM
Screen Shot 2021-01-05 at 3 56 49 PM

File: package.json [hunk 1]
```diff
@@ -152,7 +152,19 @@
           },
           "default": [],
           "description": "Per-workspace list of module directories for the language server to exclude"
-        }
+        },
+        "terraform-ls.experimentalFeatures": {
+          "scope": "resource",
```
Comment (appilon) [inline package.json:157]: Looks great overall, something I don't know well enough yet is the scopes, it's documented here

https://code.visualstudio.com/api/references/contribution-points

is `resource` the one we want?

**Review by appilon — APPROVED** [review]
Overall it LGTM but can we double check how scopes work?

**Comment by appilon** [comment]
I can confirm though the setting works!

**Comment by aeschright** [comment]
`window` would be the other option that could fit, but `resource` seems the most flexible so that's what I went with. It matches how we've done the rootModule config as well.

**Review by appilon — APPROVED** [review]

**Pull request merged**